src/components/Toolbar/Toolbar.tsx

```diff
@@ -1,27 +1,70 @@
 import React from 'react';
 import { useTranslation } from '../../hooks/useTranslation/useTranslation';
+import { AnswerModeType } from '../../types/types';
 
 type ToolbarProps = {
   title: string;
-  enableSettings: boolean;
-  showSettings: boolean;
-  toggleShowSettings: () => void;
+  activeAnswerMode: AnswerModeType;
+  enableAnswerMode: boolean;
+  enableLanguageMode: boolean;
+  handleAnswerModeChange: () => void;
+  handleLanguageModeChange: () => void;
 };
 
 export const Toolbar: React.FC<ToolbarProps> = ({
   title,
-  enableSettings,
-  showSettings,
-  toggleShowSettings,
+  activeAnswerMode,
+  enableAnswerMode,
+  enableLanguageMode,
+  handleAnswerModeChange,
+  handleLanguageModeChange,
 }) => {
   const { t } = useTranslation();
 
+  const enableTools = enableAnswerMode || enableLanguageMode;
+
+  const getSelectedAnswerMode = (): string => {
+    if (activeAnswerMode === AnswerModeType.FillIn) {
+      return 'h5p-vocabulary-drill-fill-in';
+    }
+    return 'h5p-vocabulary-drill-drag-text';
```

Comment:
Another way to do this is by matching:

```suggestion
    const classes = {
      [AnswerModeType.FillIn]: 'h5p-vocabulary-drill-fill-in',
      [AnswerModeType.DragText]: 'h5p-vocabulary-drill-drag-text',
    };

    return classes[activeAnswerMode];
```

Also, by adding a parameter to the function, it's quite easy to move it out of the component, and by extension out of the render function. Re-creating a function is in general kind of wasteful, even though in this case it wouldn't matter much.

Reply:
Better now?